### PR TITLE
Update field display in grid layout

### PR DESCRIPTION
**Pull request by jameskoster**
## What?
This PR attempts to update the display of fields in data views Grid layout, including an option for consumers to specify the orientation of appearance; column or row. 

## Why?
Currently only the field values are visible. A tooltip reveals the field name on hover, but this isn't the most accessible approach. Additionally, this pattern means you cannot always tell which field a value refers to at a glance. For instance a timestamp could be interpreted several ways. 

By arranging the field name and value in a row we can display the necessary information without consuming additional space. An option to arrange in a column means fields with much longer values (e.g. template description) can be displayed in a readable manner.

This change also moves us closer to the arrangement of data we find in the 'Details panel' and latest [Inspector designs](https://github.com/WordPress/gutenberg/issues/59689).

In the graphic below see before/after examples for templates, pages, and patterns.

fields

## How?
In `packages/dataviews/src/view-grid.js` the the `Flex` component that displays the field will switch between `row` (default) and `column` arrangements based on whether the the consumer configures a `gridDisplayDirection` property on the field. 

To ensure rows take up minimal space the value will truncate accordingly. Consumers can use the `column` orientation when more space is required.

## Testing Instructions
1. Open the templates data view, switch to grid layout and toggle on all fields
2. Notice the description is arranged as a column, and author is arranged as a row
3. When fields are displayed as a row check the value truncates correctly (you may need to manually insert a longer value via the browser inspector)

## Notes
As a follow-up it might be nice to allow consumers to say that certain field name are omitted from display. This can be useful when the name can be inferred from the value, e.g. the template description.

### CODE REVIEW

**Comment by jasmussen** [comment]
Makes sense as a step forward. A few small things:

misalignment across

Should we try to align across? Or are those flexible values?

Should we avoid the icon here?
icon next to author

We should use sentence case here:

sync status in title case

We should also have a minimum gap-width between columns, otherwise we'd probably need a colon after text items in the left column.

**Comment by jameskoster** [comment]
Oh good spot, they are actually aligned, I think I took two of the screenshots before the final tweak:

Screenshot 2024-03-22 at 12 18 45

Label is 35%, value fills remaining space after 4px gap. This felt like a reasonable balance but obviously we can tweak.

> Should we avoid the icon here?

Potentially, yes. It throws the alignment off. Probably one to discuss separately as it would affect other layouts.

> We should use sentence case here

Agree. Though I'd also note that sync status is one of those fields where the purpose _can_ be inferred from the value alone. I'd like to follow-up with a treatment for such fields, potentially a badge like we've been exploring for media:


media

**Review by jasmussen — APPROVED** [review]

File: packages/edit-site/src/components/page-templates-template-parts/index.js
Comment (ntsekouras) [inline packages/edit-site/src/components/page-templates-template-parts/index.js]: Can we wait a bit before merging this? I'm AFK today, but saw the approval and here we introduce an API in fields tied to a specific view. We need to think this better and it's time to find a better solution that also covers the `width, min-width` props we have, that are similar cases(they are tied to table view).

Comment (jameskoster) [reply packages/edit-site/src/components/page-templates-template-parts/index.js]: For sure, I won't merge until you've reviewed.

Comment (ntsekouras) [reply packages/edit-site/src/components/page-templates-template-parts/index.js]: Thank you! I pushed an update to move this option to layout config.

File: packages/dataviews/src/view-grid.js
Comment (ntsekouras) [inline packages/dataviews/src/view-grid.js:123]: Is there a specific reason for adding this inline?

**Review by ntsekouras — APPROVED** [review]
This LGTM, thanks!

File: packages/dataviews/src/view-grid.js
Comment (jameskoster) [reply packages/dataviews/src/view-grid.js:123]: Good question. It's because `Flex` seems to add an inline `height: 100%` style when the `direction` is `column`, which produces inconsistent spacing in this context. In the screenshot below note that the gap beneath the description field is different in these adjacent grid items:

Screenshot 2024-03-26 at 11 30 29
Screenshot 2024-03-26 at 11 30 19

`height: auto` fixes this. It needs to be inline to override the inherent style.

**Pull request merged**
